### PR TITLE
test(core): add unit tests for deriveServerKey and isValidServerKey

### LINKED ISSUE #1708: Add tests for mcp-utils deriveServerKey and isValidServerKey
## Summary

The mcp-utils.ts file contains `deriveServerKey` and `isValidServerKey` functions that lack test coverage.

## Files

- `packages/core/src/mcp-utils.ts`
- `packages/core/src/mcp-utils.test.ts` (to create)

## Task

Write unit tests covering:
- `deriveServerKey`: various URL formats, edge cases, invalid inputs
- `isValidServerKey`: valid keys, invalid keys, boundary cases

## Verification

- [ ] `npm run check-types` passes
- [ ] `npm run lint` passes
- [ ] `npm test` passes
- [ ] New tests provide good coverage of both functions

### PR DESCRIPTION
**Pull request by MethodManav**
## Summary
- Fixes #1708 

This PR adds comprehensive unit test coverage for the `deriveServerKey` and `isValidServerKey` utilities in `mcp-utils.ts`, which previously lacked tests.

## Changes

- Added `packages/core/src/mcp-utils.test.ts`
- Added test coverage for **`deriveServerKey`**, including:
  - Common URL formats (`https`, subdomains, ports, paths)
  - Known prefix filtering (`api`, `mcp`, `app`, etc.)
  - Single and multi-part TLD handling (`.com`, `.co.uk`, `.com.au`)
  - Edge cases such as:
    - `localhost`
    - IP addresses
    - Uppercase hostnames
    - Invalid or malformed URL strings
    - Empty input
  - Ensuring derived keys are normalized and lowercased

- Added test coverage for **`isValidServerKey`**, including:
  - Valid keys (alphanumeric, underscores, trimmed input)
  - Invalid keys (too short, empty, spaces, special characters)
  - Boundary cases (minimum valid length)
  
  
Screenshot 2026-01-21 at 10 23 46 PM

### CODE REVIEW

File: packages/core/src/mcp-utlis.test.ts [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+  });
+
+  describe("edge cases", () => {
+    it("handles localhost", () => {
+      expect(deriveServerKey("http://localhost")).toBe("localhost");
+    });
+
+    it("handles IP addresses", () => {
+      expect(deriveServerKey("http://127.0.0.1")).toBe("0");
+    });
```
Comment (lachieh) [inline packages/core/src/mcp-utlis.test.ts:69]: Let's adjust this so that it the result becomes 127_0_0_1. If you update the test first, then updating the `deriveServerKey` function will match the test.

- `http://127.0.0.1` > `127_0_0_1`
- `http://255.255.255.254` > `255_255_255_254`

Comment (MethodManav) [reply packages/core/src/mcp-utlis.test.ts:69]: I have updated the tests and refactored the `deriveServerKey` function. All tests are now passing.
